src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -79,4 +123,91 @@ public UserTenantCollection getByUsernameAndTenantId(UUID consortiumId, String u
     result.setTotalRecords(1);
     return result;
   }
+
+  private User prepareShadowUser(UUID userId, UserTenant userTenantDto) {
+    User user = new User();
+    User userOptional = getUser(userId);
+
+    if (Objects.nonNull(userOptional.getId())) {
+      user.setId(UUID.randomUUID().toString());
+      user.setPatronGroup(PATRON_GROUP);
+      user.setUsername(userOptional.getUsername() + HelperUtils.randomString(RANDOM_STRING_COUNT));
+      if(Objects.nonNull(userOptional.getPersonal())) {
+        Personal personal = new Personal();
+        personal.setLastName(userOptional.getPersonal().getLastName());
+        personal.setFirstName(userOptional.getPersonal().getFirstName());
+        personal.setEmail(userOptional.getPersonal().getEmail());
+        personal.setPreferredContactTypeId(userOptional.getPersonal().getPreferredContactTypeId());
+        user.setPersonal(personal);
+      }
+      user.setPatronGroup(userOptional.getPatronGroup());
+      user.setActive(true);
+    } else {
+        throw new ResourceNotFoundException(USER_ID, userId.toString());
+    }
+    prepareContextForTenant(userTenantDto.getTenantId());
+    return user;
+  }
+
+  private void createOrUpdateShadowUser(UUID userId, User shadowUser) {
+    User user = getUser(userId);
+    if(Objects.nonNull(user.getActive())) {
+      updateUser(user);
+    } else {
+        createUser(shadowUser);
+    }
+  }
+
+  private User getUser(UUID userId) {
+    try {
+      log.info("Getting user by userId {}.", userId);
+      return usersClient.getUsersByUserId(String.valueOf(userId));
+    } catch (FeignException e) {
+        log.info("User with userId {} does not exist in schema.", userId);
```

Comment:
probably should be a debug statement since this exception would happen often.

Reply:
fixed